src/mock.rs

```diff
@@ -1,41 +1,46 @@
 /*!
 Mock module to ease testing
     */
-use crossbeam_channel::{bounded, Receiver};
+use async_channel::{bounded, Receiver};
+use async_std::sync::Mutex;
+use async_trait::async_trait;
+use futures::future;
 
 use crate::response::Response;
-use crate::sender::Sender;
+use crate::sender::{Sender, StopFuture};
 use crate::transmission::Options;
 use crate::Event;
 use crate::Result;
 
 /// Transmission mocker for use in tests (mostly in beeline-rust)
-#[derive(Debug, Clone)]
+#[derive(Debug)]
 pub struct TransmissionMock {
     started: usize,
     stopped: usize,
     events_called: usize,
-    events: Vec<Event>,
+    events: Mutex<Vec<Event>>,
```

Comment:
Ah this is one where without `Clone` (and with this `Mutex` here, we pass on all the complexity to consumers).

I'm not saying its the wrong thing to do but the expectation that we'd manage this on the whole is now void.

Reply:
I'm still re-familiarizing myself with this PR and digesting the `Clone` issue for `Client`, but isn't this particular comment just on the testing mock?

Reply:
It is indeed. I use this in `beeline-rust` (to test multi threads) https://github.com/nlopes/beeline-rust/blob/master/src/lib.rs#L211